### PR TITLE
feat: add docs for tls replication

### PR DESCRIPTION
**Pull request by kostasrim**

### CODE REVIEW

**Comment by kostasrim** [comment]
question, I think I should ignore the `yarn.lock` and not include in the commit (which I already did). If I am wrong here, plz correct me and I will add it.

File: docs/managing-dragonfly/replication.md [hunk 1]
```diff
@@ -39,3 +39,88 @@ replicaof no one
...
+
+To enable TLS for dragonfly, you must supply the following arguments:
+
+```bash
+./dragonfly --tls --tls_key_file=server-key.pem --tls_cert_file=server-cert.pem --tls_ca_cert_file=ca-cert.pem --tls_replication=true
+```
+
+Start the replica on a different port (in this example the port is 6380, and the host is local host):
+```bash
+./dragonfly --tls --tls_key_file=server-key.pem --tls_cert_file=server-cert.pem --tls_ca_cert_file=ca-cert.pem --tls_replication=true --port 6380
```
Comment (adiholden) [inline docs/managing-dragonfly/replication.md:91]: I think its better to give an example where master and replica dont run on the same vm
use port 6379 for both master and replica

**Comment by royjacobson** [comment]
I'd like to delay this for a bit until we implement the changes from my security document.

File: docs/managing-dragonfly/replication.md [hunk 2]
```diff
@@ -39,3 +39,88 @@ replicaof no one
 
 This will stop the instance from replicating the primary server but will not discard the dataset it has already replicated. This syntax is useful in cases where the original primary fails. After running `replicaof no one` on a replica of the failed primary, the former replica can be used as the new primary and have its own replicas as a failsafe.
 
+
+## Secure replication with TLS -- Prerequisites
+
+Dragonfly supports replication over TLS. To make it work you need:
+
+1. A valid certificate issued by a CA. We recommend [let's encrypt](https://letsencrypt.org/).
```
Comment (royjacobson) [inline docs/managing-dragonfly/replication.md:47]: This is confusing to me. What's the role of the public-CA-signed cert? Which one is it?

File: docs/managing-dragonfly/replication.md [hunk 3]
```diff
@@ -39,3 +39,88 @@ replicaof no one
...
+
+Start the replica on a different port (in this example the port is 6380, and the host is local host):
+```bash
+dragonfly --tls --tls_key_file=client-key.pem --tls_cert_file=client-cert.pem --tls_ca_cert_file=ca-cert.pem --tls_replication=true --port 6380
+```
+
+Finally, connect to the replica and issue a `REPLICAOF` command:
+
+```bash
+redis-cli --tls --key ./client-key.pem --cert ./client-cert.pem --cacert ./ca-cert.pem -p 6380
```
Comment (adiholden) [inline docs/managing-dragonfly/replication.md:96]: rename cliet-key

File: docs/managing-dragonfly/replication.md [hunk 4]
```diff
@@ -39,3 +39,88 @@ replicaof no one
...
+you should always consider encrypting private keys.
+
+Finally, you should use the CA's private key to sign dragonfly's CSR and get back the signed certificate
+```bash
+openssl x509 -req -in server-req.pem -days 365 -CA ca-cert.pem -CAkey ca-key.pem -CAcreateserial -out server-cert.pem
+```
+
+The `ca-cert.pem` and `ca-key.pem` can be generated like in the first step described above. The same process applies for generating client keys.
+
+When generating private keys, always be mindful about storing and provisoning them securely!
```
Comment (royjacobson) [inline docs/managing-dragonfly/replication.md:78]:
```suggestion
When generating private keys, always be mindful about storing and provisioning them securely!
```

**Pull request merged**